### PR TITLE
matmul numpy

### PR DESCRIPTION
**Pull request by ShiftLove**

### CODE REVIEW

**Review by rush2406 — CHANGES_REQUESTED** [review]
Hi @ShiftLove 
Thanks for contributing! Could you please mention the issue created for this task?
If I understand you're trying to implement the numpy frontend for matmul? Could you please refer to this [guide](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#frontend-apis) for implementation. Let me know in case of any questions!

**Comment by ShiftLove** [comment]
> Hi @ShiftLove Thanks for contributing! Could you please mention the issue created for this task? If I understand you're trying to implement the numpy frontend for matmul? Could you please refer to this [guide](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#frontend-apis) for implementation. Let me know in case of any questions!

Hi @rush2406 , Yes I am trying to implement the numpy frontend for matmul. The issue linked with the PR is https://github.com/unifyai/ivy/issues/3624

**Review by rush2406 — CHANGES_REQUESTED** [review]
A separate is not required to be created, rather can you please shift the function to [this path](https://github.com/unifyai/ivy/blob/c1d9685561e0856e3349cfff08ababf5f3b559f1/ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py). Also please add the corresponding test function [here](https://github.com/unifyai/ivy/blob/c1d9685561e0856e3349cfff08ababf5f3b559f1/ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_linalg/test_matrix_and_vector_products.py). You can refer [example here](https://github.com/unifyai/ivy/blob/c1d9685561e0856e3349cfff08ababf5f3b559f1/ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_linalg/test_decompositions.py#L14).

The docstrings and examples are not required

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# matmul
+@handle_cmd_line_args
```
Comment (rush2406) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:2]: can you please add the necessary imports in this file?

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        available_dtypes=ivy_np.valid_float_dtypes,
+        min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=2,
+        max_dim_size=5,
+    ),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.numpy.linalg.matmul"
+    ),
+    fm=st.booleans(),
```
Comment (rush2406) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:14]: this is incorrect, please check the example mentioned below

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        fw=fw,
+        frontend="numpy",
+        fn_tree="linalg.matmul",
+        rtol=1e-02,
+        a=np.array(x, dtype=dtype),
+    )
+    
+  
+   
+Footer
```
Comment (rush2406) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:43]: remove this line please

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    x2: array,
+    /,
+    out=None,
+    *,
+    casting='same_kind',
+    order='K',
+    dtype=None,
+    subok=True
+):
+    helpers.frontend_function(
```
Comment (rush2406) [inline ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:28]: this is where the implementation using ivy should be defined, you can refer [example](https://github.com/unifyai/ivy/blob/6f27c45a4deef5ec2ad1e5117ef2937ce6f99943/ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py#L68) and perform the changes accordingly

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_linalg/test_matrix_and_vector_products.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        input_dtypes=[dtype],
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="numpy",
+        fn_tree="linalg.matmul",
+        rtol=1e-02,
+        a=np.array(x, dtype=dtype),
```
Comment (rush2406) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_linear_algebra/test_linalg/test_matrix_and_vector_products.py:37]: matmul takes 2 inputs, so please add the arguments here as well. Please refer [example](https://github.com/unifyai/ivy/blob/6f27c45a4deef5ec2ad1e5117ef2937ce6f99943/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_np_arithmetic_operations.py#L204) and do the changes accordingly

**Review by rush2406 — CHANGES_REQUESTED** [review]
please see the comments below. Also please delete the matmul file it's not required

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py (hunk 1 above)
Comment (ShiftLove) [reply ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:2]: Added

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py (hunk 3 above)
Comment (ShiftLove) [reply ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:43]: Removed

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py (hunk 4 above)
Comment (ShiftLove) [reply ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:28]: Done

File: ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py (hunk 2 above)
Comment (ShiftLove) [reply ivy/functional/frontends/numpy/linear_algebra/matrix_and_vector_products.py:14]: Done

**Comment by ShiftLove** [comment]
> please see the comments below. Also please delete the matmul file it's not required

All changes made as request. Thanks for your guidance and patience. I have learnt a lot! @rush2406 Let me know of any additional changes!

**Comment by ShiftLove** [comment]
@rush2406 There were conflicts in this PR with the Master branch which have since been resolved. I however think this action has closed the PR. I cant seem to reopen it. Can it be reopened from your side?

**Pull request closed without merging**